multicast/src/main/kotlin/com/dropbox/flow/multicast/ChannelManager.kt

```diff
@@ -75,12 +77,24 @@ internal class ChannelManager<T>(
         when (msg) {
             is Message.AddChannel -> doAdd(msg)
             is Message.RemoveChannel -> doRemove(msg.channel)
-            is Message.DispatchValue -> doDispatchValue(msg)
-            is Message.DispatchError -> doDispatchError(msg)
-            is Message.UpstreamFinished -> doHandleUpstreamClose(msg.producer)
+            is Message.Dispatch.Value -> doDispatchValue(msg)
+            is Message.Dispatch.Error -> doDispatchError(msg)
+            is Message.Dispatch.UpstreamFinished -> doHandleUpstreamClose(msg.producer)
```

Comment:
hmm i thought Dispatch was intended to be about value / error but upstream Finished is also there is a bit weird.
Maybe it is is fine, just as a note.
It almost feels like:
Miessage.Dispatch is all messages coming from Upstream

Maybe thats the the naming we can use ? (Message.Downsteam (add,remove) vs Message.Upstream (value, error, finished)) ?

Reply:
finish is also a type of dispatch IMH, but happy to change it to upstream vs downstream

Reply:
Would you mind if I make this change in a seperate PR? I don't want to mess up the stack too much

Reply:
Separate is fine